expression/collation.go

```diff
@@ -241,7 +241,16 @@ func deriveCollation(ctx sessionctx.Context, funcName string, args []Expression,
 		return ec, nil
 	case ast.Case:
 		// FIXME: case function aggregate collation is not correct.
```

Comment:
Should this FIXME still kept?(I guess yes).

Reply:
yes, this pr only handle the `then expression`, we should also deal with the `when expression`